pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -315,15 +315,15 @@ func (service *AlertRuleService) ReplaceRuleGroup(ctx context.Context, orgID int
 	})
 }
 
-// CreateAlertRule creates a new alert rule. This function will ignore any
+// UpdateAlertRule updates a new alert rule. This function will ignore any
 // interval that is set in the rule struct and fetch the current group interval
 // from database.
 func (service *AlertRuleService) UpdateAlertRule(ctx context.Context, rule models.AlertRule, provenance models.Provenance) (models.AlertRule, error) {
 	storedRule, storedProvenance, err := service.GetAlertRule(ctx, rule.OrgID, rule.UID)
 	if err != nil {
 		return models.AlertRule{}, err
 	}
-	if storedProvenance != provenance && storedProvenance != models.ProvenanceNone {
+	if storedProvenance != provenance && (storedProvenance == models.ProvenanceFile || storedProvenance == models.ProvenanceAPI && provenance == models.ProvenanceFile) {
```

Comment:
I had to read this twice to understand it. It would help to precompute this into variables and then use those in your expression.

```suggestion
    isAPItoFile := storedProvenance == models.ProvenanceAPI && provenance == models.ProvenanceFile
    isFile := storedProvenance == models.ProvenanceFile
	if storedProvenance != provenance && (isFile || isAPItoFIle ) {
```